src/ducks/components/intents/InstallAppIntent.jsx

```diff
@@ -34,32 +35,33 @@ export class InstallAppIntent extends Component {
     }
   }
 
-  async componentWillReceiveProps(nextProps) {
+  async componentDidUpdate(prevProps) {
     // on install success
-    if (this.props.isInstalling && !nextProps.isInstalling) {
-      this.setState({
-        status: 'installed'
-      })
-
+    if (
+      (prevProps.app && prevProps.app.installed) ||
+      (this.props.app && this.props.app.installed)
+    ) {
       const { app, compose, data, onTerminate } = this.props
-      const configure = typeof data.configure === 'undefined' || data.configure
+      const doConfiguration =
+        data && (typeof data.configure === 'undefined' || data.configure)
 
-      if (app.type === 'konnector' && configure) {
+      if (app.type === APP_TYPE.KONNECTOR && doConfiguration) {
         await compose(
           'CREATE',
           'io.cozy.accounts',
           { slug: app.slug }
         )
       }
 
-      onTerminate(nextProps.app)
+      onTerminate(this.props.app)
```

Comment:
`onTerminate` should be call into `AppInstallSucces` now, as it is always displayed. That why we got a termination message too early.